gems/smithy-client/lib/smithy-client/param_converter.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+require 'bigdecimal'
+require 'stringio'
+require 'date'
+require 'time'
+require 'tempfile'
+
+module Smithy
+  module Client
+    # @api private
+    class ParamConverter
+      include Smithy::Client::Shapes
+
+      @mutex = Mutex.new
```

Comment:
I assume this is all effectively copied from V3? It looks mostly identical.

Reply:
Yes mostly but I cleaned up code, added extra shape support, and extra conversions.